File: pkg/gate/testdata/ts-proof-pack/fixtures/ts/contract-sig-present.ts
// contract-sig-present.ts (TASK-002): a real .ts fixture whose declared function
// signature is PRESENT, so the shared TS proof pack's contract signature rule
// MATCHES via real ast-grep (verdict SATISFIED — CLM-023).
export function routeFile(path: string, mode: number): string {
  return path;
}

File: pkg/gate/testdata/contract-sig-present.go
package contractfix

// contract-sig-present.go (TASK-001): a real Go fixture whose declared symbol
// signature is PRESENT, so the pack's ast-grep signature rule MATCHES and the
// gate verdicts SATISFIED (CLM-004). Used by the Go pack rule tests and the
// strangler-equivalence pass on real ast-grep.

// RouteFile resolves a path to a routing decision. Its signature is the one the
// contract fixtures declare as PRESENT.
func RouteFile(path string, mode int) (string, error) {
	if path == "" {
		return "", nil
	}
	return path, nil
}
